Test runner fails when the module does not have a direct reference to zio-test

```
Testing started at 4:57 PM ...
Exception in thread "main" java.lang.NoClassDefFoundError: scala/$less$colon$less
    at zio.intellij.testsupport.ZTestRunner.main(ZTestRunner.scala)
Caused by: java.lang.ClassNotFoundException: scala.$less$colon$less
    at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
    at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:352)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
    ... 1 more
```

The problem is when a project containing tests does not have an explicit `libraryDependency` on the `zio-test` library (instead, depends on another project, e.g. `.dependsOn(shared % "compile->compile;test->test")`.

--- a/src/main/scala/zio/intellij/testsupport/runner/TestRunnerProjectNotification.scala
+++ b/src/main/scala/zio/intellij/testsupport/runner/TestRunnerProjectNotification.scala
@@ -2,13 +2,14 @@ package zio.intellij.testsupport.runner
 
 import com.intellij.ide.BrowserUtil
 import com.intellij.notification.{Notification, NotificationGroup, NotificationListener, NotificationType}
+import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.project.Project
-import javax.swing.event.HyperlinkEvent
 import org.jetbrains.annotations.NonNls
 import org.jetbrains.plugins.scala.project.{ModuleExt, ProjectExt}
 import zio.intellij.ZioIcon
-import zio.intellij.testsupport.runner.TestRunnerNotifications.displayInfo
 import zio.intellij.utils.ModuleSyntax
+
+import javax.swing.event.HyperlinkEvent
 
 private[runner] final class TestRunnerProjectNotification(private val project: Project) {
   def init(): Unit =
@@ -19,15 +20,15 @@ private[runner] final class TestRunnerProjectNotification(private val project: P
   private def versions(project: Project) = {
     val sourceModules = project.modulesWithScala.filter(_.isSourceModule).toList
 
-    sourceModules.view
+    sourceModules
       .flatMap(m => m.zioVersion zip m.scalaVersion)
-      .headOption
+      .distinct
   }
 
   private def shouldSuggestTestRunner(project: Project, downloadIfMissing: Boolean = false): Boolean =
-    versions(project).fold(false) {
-      case (version, scalaVersion) =>
-        TestRunnerResolveService.instance
+    versions(project).foldLeft(false) {
+      case (flag, (version, scalaVersion)) =>
+        flag | TestRunnerResolveService.instance
           .resolve(version, scalaVersion, downloadIfMissing)
           .toOption
           .isEmpty
@@ -48,8 +49,10 @@ private[runner] final class TestRunnerProjectNotification(private val project: P
                 scalaVersion,
                 project,
                 onResolved = {
-                  case Right(_) => displayInfo("ZIO Test runner was downloaded successfully!")
-                  case _        => // relying on error reporting in resolve method
+                  case Right(_) =>
+                    val logger = Logger.getInstance("ZIO plugin for IntelliJ")
+                    logger.info(s"ZIO Test runner for $version (${scalaVersion.major}) was downloaded successfully!")
+                  case _ => // relying on error reporting in resolve method
                 }
               )
         }

--- a/src/main/scala/zio/intellij/testsupport/runner/TestRunnerResolveService.scala
+++ b/src/main/scala/zio/intellij/testsupport/runner/TestRunnerResolveService.scala
@@ -27,7 +27,7 @@ import scala.util._
 private[testsupport] final class TestRunnerResolveService
     extends PersistentStateComponent[TestRunnerResolveService.ServiceState] {
 
-  private val testRunnerVersions: mutable.Map[Version, ResolveStatus] = ScalaCollectionsUtil.newConcurrentMap
+  private val testRunnerVersions: mutable.Map[(Version, String), ResolveStatus] = ScalaCollectionsUtil.newConcurrentMap
 
   private val state: TestRunnerResolveService.ServiceState = new TestRunnerResolveService.ServiceState
 
@@ -41,13 +41,14 @@ private[testsupport] final class TestRunnerResolveService
     downloadIfMissing: Boolean,
     resolveFast: Boolean = false,
     progressListener: DownloadProgressListener = NoopProgressListener
-  ): ResolveResult = testRunnerVersions.get(version) match {
+  ): ResolveResult = testRunnerVersions.get((version, scalaVersion.major)) match {
     case Some(ResolveStatus.Resolved(jarPaths)) => Right(jarPaths)
     case _ if resolveFast                       => Left(ResolveError.NotFound(version, scalaVersion))
     case Some(ResolveStatus.DownloadInProgress) => Left(ResolveError.DownloadInProgress(version, scalaVersion))
     case _ =>
-      if (state.resolvedVersions.containsKey(version.toString)) {
-        val jarUrls = state.resolvedVersions.get(version.toString).map(new URL(_))
+      val key = s"${version.toString}###${scalaVersion.major}"
+      if (state.resolvedVersions.containsKey(key)) {
+        val jarUrls = state.resolvedVersions.get(key).map(new URL(_))
         resolveClassPath(version, scalaVersion, jarUrls.toIndexedSeq) match {
           case r @ Right(_)                 => r
           case Left(_) if downloadIfMissing => downloadAndResolve(version, scalaVersion, progressListener)
@@ -63,7 +64,7 @@ private[testsupport] final class TestRunnerResolveService
     project: Project,
     onResolved: ResolveResult => Unit = _ => ()
   ): Unit =
-    testRunnerVersions.get(version) match {
+    testRunnerVersions.get((version, scalaVersion.major)) match {
       case Some(ResolveStatus.Resolved(fmt)) =>
         invokeLater(onResolved(Right(fmt)))
       case Some(ResolveStatus.DownloadInProgress) =>
@@ -104,8 +105,9 @@ private[testsupport] final class TestRunnerResolveService
     val urls: Array[URL] = jarUrls.toArray
     Try(new URLClassLoader(urls, null).loadClass(ZTestRunnerName + "$")) match {
       case Success(_) =>
-        state.resolvedVersions.put(version.toString, jarUrls.toArray.map(_.toString))
-        testRunnerVersions(version) = ResolveStatus.Resolved(jarUrls)
+        val key = s"${version.toString}###${scalaVersion.major}"
+        state.resolvedVersions.put(key, jarUrls.toArray.map(_.toString))
+        testRunnerVersions((version, scalaVersion.major)) = ResolveStatus.Resolved(jarUrls)
         Right(jarUrls)
       case Failure(e) =>
         Left(ResolveError.UnknownError(version, scalaVersion, e))

